### PR TITLE
Add lesson 10

### PR DESCRIPTION
**Pull request by sevtapim**

### CODE REVIEW

File: lesson10.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+Able to define functions
+
+Conditionally execute code
+
+Limited ability to interact with a user
+
+---
+
+The values we operate on are so <span style="color: indianred;">basic</span> that, any combination of values still produces one single value
```
Comment (haraldF) [inline lesson10.md]:
```suggestion
The values we operate on are so <span style="color: indianred;">basic</span> that any combination of values still produces one single value
```

File: lesson10.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+`34 * 10 + 170`
+
+String:
+
+`"Hello, I am Anne from Denmark, 34 years-old"`
+
+---
+
+The data about Anne is scattered around in multiple variables.
```
Comment (haraldF) [inline lesson10.md]:
```suggestion
The data about Anne is scattered around multiple variables:
```

File: lesson10.md [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+---
+
+A variable can hold every possible value the data type allows. JS won’t complain if we tried to set
+
+```js
+age = 10000; 
+height = -1800;
+origin = "Lala land";
+```
+
```
Comment (haraldF) [inline lesson10.md]:
```suggestion
```

File: lesson10.md [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+```js
+person1Name, person1Age, person1Height, person1Origin
+person2Name, person2Age, person2Height, person2Origin
+person3Name, person3Age, person3Height, person3Origin
+```
+
+---
+
+In producing new info from existing data, basic data types are useful when
```
Comment (haraldF) [inline lesson10.md]:
```suggestion
In producing new information from existing data, basic data types are useful when
```

File: lesson10.md [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+
+In Abdullah’s game, we can think of a collection for holding a person’s name, age and height as one single unit
+
+`[name, age, height]`
+
+---
+
+### Increasing Code’s ability
+
+<span style="color: indianred;">Collecting</span> values into one unit is so essential, that JS provides 2 different data types for representing them: <span style="color: indianred;">Object</span> and <span style="color: indianred;">Array</span>.
```
Comment (haraldF) [inline lesson10.md]:
```suggestion
<span style="color: indianred;">Collecting</span> values into one unit is so essential that JS provides 2 different data types for representing them: <span style="color: indianred;">Object</span> and <span style="color: indianred;">Array</span>.
```

File: lesson10.md [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+
+### JS Objects
+
+A JS Object collects values between two curly brackets
+
+```js
+{name: "Anne", 
+age: 34, 
+height: 170, 
+origin: "Denmark"}
```
Comment (haraldF) [inline lesson10.md]:
```suggestion
{
  name: "Anne", 
  age: 34, 
  height: 170,
  origin: "Denmark"
}
```

File: lesson10.md [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+```
+
+Every element between two curley brackets are called <span style="color: indianred;">object properties</span>
```
Comment (haraldF) [inline lesson10.md]:
```suggestion
Every element between two curly braces is called <span style="color: indianred;">object property</span>
```

File: lesson10.md [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+
+`name, age, height, and origin` are called <span style="color: indianred;">property names</span>
+
+`"Anne", 34, 170, "Denmark"` are called <span style="color: indianred;">property values</span>
+
+---
+
+A property is also called a <span style="color: indianred;">key-value pair</span>, especially useful when you view your object as a dictionary
```
Comment (haraldF) [inline lesson10.md]:
```suggestion
A property is also called a <span style="color: indianred;">key-value pair</span>. Key-value Paris are especially useful when we view an object as a dictionary
```

File: lesson10.md [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+```
+
+---
+
+### Object
+
+A collection has a data type: <span style="color: indianred;">Object</span>
+
+```js
+typeof {name: "Anne", age: 34, height: 170} // object
```
Comment (haraldF) [inline lesson10.md]:
```suggestion
typeof {name: "Anne", age: 34, height: 170} // "object"
```

File: lesson10.md [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+
+```js
+{age: 34}
+```
+
+---
+
+### Assign to a variable
+
+And how we <span style="color: cadetblue;">assign an object to a variable</span>:
```
Comment (haraldF) [inline lesson10.md]:
```suggestion
And here is how we <span style="color: cadetblue;">assign an object to a variable</span>:
```

File: lesson10.md [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+
+```js
+let anne = {age: 34};
+```
+
+---
+
+### Read a property’s value
+
+And how we <span style="color: cadetblue;">read the value back from the object</span>:
```
Comment (haraldF) [inline lesson10.md]:
```suggestion
Here is how we <span style="color: cadetblue;">read the value of a property from the object</span>:
```

File: lesson10.md [hunk 12]
```diff
@@ -0,0 +1,304 @@
...
+
+```js
+anne.age = 35;
+```
+
+---
+
+### Object data type
+
+When working with Object data type, the order does not matter
```
Comment (haraldF) [inline lesson10.md]:
```suggestion
When working with Object data type, the order of the properties does not matter
```

File: lesson10.md [hunk 13]
```diff
@@ -0,0 +1,304 @@
...
+### Methods
+
+A property's value can also be a function
+
+```js
+{
+    name: "Ekas",
+    greet: function() {
+        return "Hello, I am " + this.name;
+        }
```
Comment (haraldF) [inline lesson10.md]:
```suggestion
    }
```

File: lesson10.md [hunk 14]
```diff
@@ -0,0 +1,304 @@
...
+
+### Methods
+
+A property's value can also be a function
+
+```js
+{
+    name: "Ekas",
+    greet: function() {
+        return "Hello, I am " + this.name;
```
Comment (haraldF) [inline lesson10.md]: I think we need an attritional slide to introduce the "this" variable

File: lesson10.md (hunk 2 above)
Comment (sevtapim) [reply lesson10.md]: - We have multiple variables to hold data about Anne.

**Pull request merged**